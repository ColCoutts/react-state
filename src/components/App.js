import React from 'react';
import RandomColor from './RandomColor';

export default function App() {
  return <RandomColor />;
}

// export default class App extends PureComponent {
//   state = {
//     color: '',
//     starter: new Date(),
//     img: ''
//   }

//   componentDidMount() {
//     this.starterTime = setInterval(() => {
//       this.setState({ color: this.randomHex(), starter: new Date(), img: '' });
//     }, 1000);
//   }

//   componentDidUpdate(prevProps, prevState) {
//     if(prevState === this.state.color) {
//       this.setState({ img: 'https://thumbs.gfycat.com/DemandingWelltodoGallinule-poster.jpg' });
//     }
//   }

//   componentWillUnmount() {
//     this.starterTime && clearInterval(this.starterTime);
//   }

//   render() {
//     const { color } = this.state;
//     const styles = {
//       background: color,
//       width: '40em',
//       height: '40em'
//     };

//     return <RandomColor style={styles} />;
//   }

// }
