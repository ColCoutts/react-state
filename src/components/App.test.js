import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App tests', () => {
  it('renders an App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('changes color on selectColor even', () => {
    const wrapper = shallow(<App />);

    wrapper.instance().selectColor('green');
    expect(wrapper.state('color')).toEqual('green');
  });
});
