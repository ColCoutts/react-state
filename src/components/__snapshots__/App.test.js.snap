// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests renders an App 1`] = `
<Fragment>
  <ColorPicker
    selectColorHandler={[Function]}
  />
  <ColorDisplay
    color=""
  />
</Fragment>
`;
